src/DeterministicVI.jl

```diff
@@ -66,19 +68,17 @@ function ElboArgs{NumType <: Number}(
     @assert length(tile_source_map) == N
     @assert size(patches, 1) == S
     @assert size(patches, 2) == N
-    for tiled_image in images
-        for tile in tiled_image.tiles
-            for ep in tile.epsilon_mat
-                if ep <= 0.0
-                    throw(InvalidInputError(
-                        "You must set all values of epsilon_mat > 0 for all images included in ElboArgs"
-                    ))
-                end
-            end
+
+    for img in images, tile in img.tiles, ep in tile.epsilon_mat
```

Comment:
@gostevehoward -- I changed the epsilon check to loop over the values. It turns out that `all(isnan...)` allocates a lot of memory, I guess it's forming any array of booleans for each pixel before running `all`.